message_adapter/cumulus_message.py

```diff
@@ -0,0 +1,263 @@
+        return resolve_path_str(event, input_path)
+    return event.get('payload')
+
+
+def resolve_config_templates(event, config):
+    """
+    * Given a config object containing possible JSONPath-templated values, resolves
+    * all the values in the object using JSONPaths into the provided event.
+    *
+    * @param {*} event The event that paths resolve against
+    * @param {*} config A config object, containing paths
+    * @returns {*} A config object with all JSONPaths resolved
+    """
+    task_config = config.copy()
+    if 'cumulus_message' in task_config:
+        del task_config['cumulus_message']
+    return _resolve_config_object(event, task_config)
+
+
+def store_remote_response(incoming_event, max_size, config_keys):
+    """
+    * Stores part of a response message in S3 if it is too big to send to StepFunctions
+    * @param {*} incoming_event  - The response message
+    * @param {*} max_size        - The maximum size (in bytes) a response message portion
+    *                              can be before the method will store it in s3
+    * @param {*} config_keys     - A list of valid CMA configuration keys
+    * @returns {*} A response message, possibly referencing an S3 object for its contents
+    """
+    event = deepcopy(incoming_event)
+    replace_config = event.get('ReplaceConfig', None)
+    if not replace_config:
+        return event
+    # Set default value if FullMessage flag set
+    if replace_config.get('FullMessage', False):
+        replace_config['Path'] = '$'
+
+    replace_config_values = _parse_remote_config_from_event(replace_config, max_size)
+
+    for key in config_keys:
+        if event.get(key):
+            del event[key]
+    cumulus_meta = deepcopy(event['cumulus_meta'])
+    replacement_data = replace_config_values['parsed_json_path'].find(event)
+    if len(replacement_data) != 1:
+        raise Exception(f'JSON path invalid: {replace_config_values["parsed_json_path"]}')
+    replacement_data = replacement_data[0]
+
+    estimated_data_size = len(json.dumps(replacement_data.value))
+    if sys.version_info.major > 3:
+        estimated_data_size = len(json.dumps(replacement_data.value).encode(('utf-8')))
+
+    if estimated_data_size < replace_config_values['max_size']:
+        return event
+
+    _s3 = s3()
+    s3_bucket = event['cumulus_meta']['system_bucket']
+    s3_key = ('/').join(['events', str(uuid.uuid4())])
+    s3_params = {
+        'Expires': datetime.utcnow() + timedelta(days=7),  # Expire in a week
+        'Body': json.dumps(replacement_data.value)
+    }
+    _s3.Object(s3_bucket, s3_key).put(**s3_params)
+
+    try:
+        replacement_data.value.clear()
+    except AttributeError:
+        replace_config_values['parsed_json_path'].update(event, '')
+
+    remote_configuration = {'Bucket': s3_bucket, 'Key': s3_key,
+                            'TargetPath': replace_config_values['target_path']}
+    event['cumulus_meta'] = event.get('cumulus_meta', cumulus_meta)
```

Comment:
It looks like we do a `deepcopy` of `cumulus_meta` but then never actually mutate the copy? Do we need this statement and the only on L161 still?

Reply:
We *do* mutate event, and therefore possibly cumulus meta,
 however when we do a replacement_data.value.clear().  This deepcopy is preserving the cumulus_meta, regardless of what the remote message json path is.